File: RepoTransBench/target_projects/Java/C#/Cybereason_Logout4Shell/public_tests/Log4jPublicTest.cs
using System;
using System.IO;
using Xunit;
using Cybereason_Logout4Shell;

namespace PublicTests
{
    public class Log4jPublicTest : IDisposable
    {
        private StringWriter outWriter;
        private StringWriter errWriter;
        private TextWriter originalOut;
        private TextWriter originalErr;

        public Log4jPublicTest()
        {
            originalOut = Console.Out;
            originalErr = Console.Error;
            outWriter = new StringWriter();
            errWriter = new StringWriter();
            Console.SetOut(outWriter);
            Console.SetError(errWriter);
            Log4j.ThreadContext.Clear();
        }

        public void Dispose()
        {
            Console.SetOut(originalOut);
            Console.SetError(originalErr);
            Log4j.ThreadContext.Clear();
        }

        [Fact]
        public void TestConfigureLoggerWithThreadContext()
        {
            Log4j.configureLoggerWithThreadContext();
            Log4j.ThreadContext["header"] = "PUBLIC_HEADER";
            // Write error to output as in test
            Console.WriteLine($"{Log4j.ThreadContext["header"]}: Another error!");
            string output = outWriter.ToString();
            Assert.Contains("PUBLIC_HEADER", output);
            Assert.Contains("Another error!", output);
        }

        [Fact]
        public void TestMainWithThreadLocalAttack()
        {
            string[] args = new[] { "-t" };
            Log4j.main(args);
            string output = outWriter.ToString();
            Assert.Contains("Will use ThreadContext as attack vector", output);
            Assert.Contains("Vulnerable through thread context - 1", output);
            Assert.Contains("Vulnerable through thread context - 2", output);
        }

        [Fact]
        public void TestMainWithoutThreadLocalAttack()
        {
            string[] args = Array.Empty<string>();
            Log4j.main(args);
            string output = outWriter.ToString();
            // Check for presence of "ldap://127.0.0.1:1389" and "${jndi:" as in the Java public test
            Assert.Contains("ldap://127.0.0.1:1389", output);
            Assert.Contains("${jndi:", output);
        }
    }
}

File: RepoTransBench/target_projects/Java/Go/Cybereason_Logout4Shell/public_tests/log4j_public_test.go
package public_tests

import (
	"bytes"
	"strings"
	"testing"

	"cybereason_logout4shell/tests"
)

func TestLog4jDefaultRunPublic(t *testing.T) {
	var out bytes.Buffer
	log4j := &tests.Log4j{}
	log4j.Main([]string{}, &out)

	result := out.String()
	if !strings.Contains(result, "${jndi:ldap://127.0.0.1:1389/a}") {
		t.Fatalf("Public: Expected JNDI payload in default output, got: %s", result)
	}
}

func TestLog4jThreadContextPublic(t *testing.T) {
	var out bytes.Buffer
	log4j := &tests.Log4j{}
	args := []string{"-t"}
	log4j.Main(args, &out)
	result := out.String()

	if !strings.Contains(result, "Will use ThreadContext as attack vector") {
		t.Errorf("Public: Expected ThreadContext attack vector info, got: %s", result)
	}

	if !strings.Contains(result, "[ATTACK_HEADER] Vulnerable through thread context - 1") {
		t.Errorf("Public: Expected ThreadContext error line 1, got: %s", result)
	}

	if !strings.Contains(result, "[ATTACK_HEADER] Vulnerable through thread context - 2") {
		t.Errorf("Public: Expected ThreadContext error line 2, got: %s", result)
	}
}